stream-chat-android-client/src/main/java/io/getstream/chat/android/client/models/Attachment.kt

```diff
@@ -40,9 +40,28 @@ public data class Attachment(
 
 ) : CustomObject {
 
+    /**
+     * Represents various states in attachment upload lifecycle.
+     */
     public sealed class UploadState {
-        public object InProgress : UploadState()
+        /**
+         * Idle state before attachment starts to upload.
+         */
+        public object Idle : UploadState()
+
+        /**
+         * State representing attachment upload progress.
+         */
+        public data class InProgress(val bytesUploaded: Long, val totalBytes: Long) : UploadState()
```

Comment:
I'm gonna nitpick here, it's weird that we have `bytesUploaded` and then `totalBytes`.

It should either be `bytesUploaded` and `bytesTotal` (sounds awkward) or `uploadedBytes` and `totalBytes` (more natural imo). We should also sync it with the param names in `ProgressCallback.

Reply:
Actually, the `bytesUploaded` and `totalBytes` are more natural and verbal. The first one tells us the number of bytes uploaded and the second one total bytes. And this is the general convention I have seen used across many projects. 

1. https://github.com/apache/eagle/blob/df0e91de65cf9428f0f735065eba580e37183030/eagle-core/eagle-app/eagle-app-base/src/main/java/org/apache/eagle/app/environment/impl/StormSubmitter.java#L310
2. https://github.com/square/okhttp/blob/master/samples/guide/src/main/java/okhttp3/recipes/Progress.java#L124
4. https://github.com/cloudinary/cloudinary_kotlin/blob/03df38fc5334d0acb2e8c2b5a01cce294842a21f/uploader/src/main/kotlin/com/cloudinary/http/HttpClient.kt#L5

Reply:
Alright, we can keep these names then. Let's make them consistent in `ProgressCallback`, and then this PR should be good to go.